Lib9c/Action/RankingBattle.cs

```diff
@@ -78,36 +73,39 @@ public override IAccountStateDelta Execute(IActionContext context)
             var enemyAvatarState = states.GetAvatarState(EnemyAddress);
             if (enemyAvatarState is null)
             {
-                return LogError(
-                    context,
-                    "Aborted as the avatar state of the opponent ({OpponentAddress}) was failed to load.",
-                    EnemyAddress
-                );
+                throw new FailedLoadStateException($"Aborted as the avatar state of the opponent ({EnemyAddress}) was failed to load.");
             }
 
             var weeklyArenaState = states.GetWeeklyArenaState(WeeklyArenaAddress);
-
-            //FIXME 오류던지게 고쳐야함
+            
             if (weeklyArenaState.Ended)
             {
-                return LogError(context, "Aborted as the weekly arena state already ended.");
+                throw new WeeklyArenaStateAlreadyEndedException();
             }
 
             if (!weeklyArenaState.ContainsKey(AvatarAddress))
             {
-                return LogError(context, "Aborted as the weekly arena state was failed to load.");
+                throw new WeeklyArenaStateNotContainsAvatarAddressException(AvatarAddress);
             }
 
             var arenaInfo = weeklyArenaState[AvatarAddress];
 
             if (arenaInfo.DailyChallengeCount <= 0)
             {
-                return LogError(context, "Aborted as the arena state reached the daily limit.");
+                throw new NotEnoughWeeklyArenaChallengeCountException();
             }
 
             if (!arenaInfo.Active)
             {
-                FungibleAssetValue agentBalance = states.GetBalance(ctx.Signer, states.GetGoldCurrency());
+                FungibleAssetValue agentBalance = default;
+                try
+                {
+                    agentBalance = states.GetBalance(ctx.Signer, states.GetGoldCurrency());
+                }
+                catch (InvalidOperationException)
```

Comment:
(it isn't critical, but just checking)

Does `states.GetBalance()` throw `InvalidOperationException` ? according to [docs](https://docs.libplanet.io/master/api/Libplanet.Action.IAccountStateDelta.html#Libplanet_Action_IAccountStateDelta_GetBalance_Libplanet_Address_Libplanet_Currency_), it doesn't seem to throw any exceptions.

cc @planetarium/libplanet

Reply:
Found during testing. If I put an address that has never received `Balance` in `states.GetBalance()` as the first parameter, `InvalidOperationException ` occurred.